grafana/resource_role_assignment.go

```diff
@@ -0,0 +1,186 @@
+package grafana
+
+import (
+	"context"
+
+	gapi "github.com/grafana/grafana-api-golang-client"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+)
+
+func ResourceRoleAssignment() *schema.Resource {
+	return &schema.Resource{
+		Description: `
+**Note:** This resource is available only with Grafana Enterprise 9.2+.
+* [Official documentation](https://grafana.com/docs/grafana/latest/enterprise/access-control/)
+* [HTTP API](https://grafana.com/docs/grafana/latest/http_api/access_control/)
+`,
+		CreateContext: UpdateRoleAssignments,
+		UpdateContext: UpdateRoleAssignments,
+		ReadContext:   ReadRoleAssignments,
+		DeleteContext: UpdateRoleAssignments,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		Schema: map[string]*schema.Schema{
+			"role_uid": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Grafana RBAC role UID.",
+			},
+			"users": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Description: "Role assignments to users.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:        schema.TypeInt,
+							Required:    true,
+							ForceNew:    false,
+							Description: "User ID.",
+						},
+						"global": {
+							Type:        schema.TypeBool,
+							Optional:    true,
+							Default:     false,
+							ForceNew:    false,
+							Description: "States whether the assignment is available across all organizations or not.",
+						},
+					},
+				},
+			},
+			"teams": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Description: "Role assignments to teams.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:        schema.TypeInt,
+							Required:    true,
+							ForceNew:    false,
+							Description: "Team ID.",
+						},
+					},
+				},
+			},
+			"service_accounts": {
+				Type:        schema.TypeSet,
+				Optional:    true,
+				Description: "Role assignments to service accounts.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:        schema.TypeInt,
+							Required:    true,
+							ForceNew:    false,
+							Description: "Service account ID.",
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func ReadRoleAssignments(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*client).gapi
+	uid := d.Get("role_uid").(string)
```

Comment:
A read function should always depend solely on the ID. If the role UID is the ID of the resource then: 
- That means that there can only be one role assignment per role. Is that OK?
- The Read function should start with `uid := d.GetID()` and all attributes should be set from the subsequent return of the API

This is needed for import to work. What `terraform import` does is it sets the ID and then triggers a read on the resource to fill in the details